src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs

```diff
@@ -7,131 +7,271 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
 
 	public class S3Repository : IS3Repository
 	{
-		public S3Repository(S3RepositorySettings settings)
-		{
-			Settings = settings;
-		}
+		public S3Repository(S3RepositorySettings settings) => Settings = settings;
 
 		public IS3RepositorySettings Settings { get; set; }
 		public string Type { get; } = "s3";
 	}
 
 	public interface IS3RepositorySettings : IRepositorySettings
 	{
+		/// <summary>
+		/// The name of the bucket to be used for snapshots. This field is required
+		/// </summary>
 		[JsonProperty("bucket")]
 		string Bucket { get; set; }
 
+		/// <summary>
+		/// The region where bucket is located. Defaults to US Standard.
+		/// </summary>
 		[Obsolete("Removed in NEST 6.x.")]
 		[JsonProperty("region")]
 		string Region { get; set; }
 
+		/// <summary>
+		/// The endpoint to the S3 API. Defaults to AWS’s default S3 endpoint. Note that setting a region overrides the endpoint setting.
```

Comment:
nitpick: change documentation apostrophe `’` to `'`

Reply:
Copy pasta from docs page :) will fix.